src/server.test.js

```diff
@@ -1,81 +1,74 @@
+jest.mock('./routes/userRoutes')
 jest.mock('express')
 jest.mock('../sequelize/db')
 
 const express = require('express')
 const dbModule = require('../sequelize/db')
-const {getApp, startServer, stopServer} = require('./server')
+const userRoutes = require('./routes/userRoutes')
+userRoutes.createUser = jest.fn()
+userRoutes.loginUser = jest.fn()
+userRoutes.deleteUser = jest.fn() // userRoutes should be mocked before requiring server
+
+const {startServer, stopServer, getApp} = require('./server')
 
 dbModule.start = jest.fn()
 dbModule.close = jest.fn()
 
+const app = {
+  use: () => {},
+  get: () => {},
+  post: jest.fn(),
+  delete: jest.fn(),
+  listen: jest.fn().mockImplementation((port, callback) => callback()),
+  name: 'Carl Sagan'
+}
+express.mockReturnValue(app)
+
```

Comment:
damn it. i also comprised the way carlos wrote it and changed my code to the reverse way